main.cpp

```diff
@@ -398,15 +406,19 @@ DoubleType& DoubleType::operator*=(const double other)
 DoubleType& DoubleType::apply(std::function<DoubleType&(double&)> doubleFunc)
 {
     if(doubleFunc)
-    doubleFunc(*doubleTypeHeap);
+    {
+        doubleFunc(*doubleTypeHeap);
```

Comment:
- [ ] return the result of calling the std::function if the std::function is valid.